handler/email.go

```diff
@@ -0,0 +1,114 @@
+package handler
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/covidtrace/jwt"
+	"github.com/covidtrace/operator/storage"
+	"github.com/covidtrace/operator/util"
+	"github.com/google/uuid"
+)
+
+type emailHandler struct {
+	bucket      storage.Bucket
+	issuer      *jwt.Issuer
+	refresh     *jwt.Issuer
+	codeLength  int
+	hashSalt    string
+	fromAddress string
+}
+
+func NewEmail(bucket storage.Bucket, jwtSigningKey []byte, iss, aud string, td, rd time.Duration) Handler {
+	issuer := jwt.NewIssuer(jwtSigningKey, iss, aud, td)
+	refresh := issuer.WithDur(rd)
+
+	return &emailHandler{
+		bucket:      bucket,
+		issuer:      issuer,
+		refresh:     refresh,
+		codeLength:  8,
+		hashSalt:    util.GetEnvVar("HASH_SALT"),
+		fromAddress: util.GetEnvVar("EMAIL_FROM_ADDRESS"),
+	}
+}
+
+func (h *emailHandler) Issuer() *jwt.Issuer {
+	return h.issuer
+}
+
+func (h *emailHandler) Refresher() *jwt.Issuer {
+	return h.refresh
+}
+
+func (h *emailHandler) Bucket() storage.Bucket {
+	return h.bucket
+}
+
+type emailReq struct {
+	Email string `json:"email"`
+}
+
+var emailKey contextKey
+
+func (h *emailHandler) Meta(ctx context.Context, r *http.Request) (context.Context, *string, *tokenMeta, error) {
```

Comment:
The `Meta` name for this function is a bit odd IMO. I would either add a comment to the Handler interface that describes what this does, or rename it to something more descriptive.

Reply:
Renamed, and simplified the gnarly context stuff by shoving a few more things in the `tokenMeta` struct.